### PR TITLE
feat: support multiple btc chain config

### PR DESCRIPTION
**Pull request by ws4charlie**
# Description

- [x] Allow multiple Bitcoin chain config in zetaclient config file.
- [x] Create multiple Bitcoin observer/signer pairs when configured.

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [x] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Release Notes

- **New Features**
  - Added support for stateful precompiled contracts, enhancing smart contract functionality.
  - Introduced a common importable `zetacored rpc` package for easier integration.
  - Implemented a staking precompiled contract to improve staking mechanisms.
  - Added support for restricted addresses in the Solana network for enhanced security.
  - Emission of events from the staking precompile for better transparency.
  - Expanded support for multiple Bitcoin chains, increasing compatibility.

- **Bug Fixes**
  - Improved error handling for Bitcoin chain configurations.

- **Tests**
  - Enhanced test coverage for `GetEVMConfig` and `GetBTCConfig` methods, ensuring robustness against edge cases.

### CODE REVIEW

File: zetaclient/config/types.go
Comment (gartnera) [inline zetaclient/config/types.go:91]: Ideally we would preserve the old field and merge it into the new map to make the upgrade a bit less painful

Comment (ws4charlie) [reply zetaclient/config/types.go:91]: The field `BitcoinConfig` is retained by commit https://github.com/zeta-chain/node/pull/2870/commits/b1ee2eb48b4db1debb851ba77fdf68819fd2feb7 to be backward compatible

File: zetaclient/config/types.go
Comment (gartnera) [inline zetaclient/config/types.go:133]: maybe derive the chain ID from `RPCParams` so it's not accidentally mismatched?

Comment (ws4charlie) [reply zetaclient/config/types.go:133]: We actually can't figure out chain ID from the `RPCParams`. Also, the only source of truth would need to come from the RPC endpoint if there IS a way to figure out.

File: zetaclient/orchestrator/orchestrator.go
Comment (ws4charlie) [reply zetaclient/orchestrator/orchestrator.go:413]: it's a renaming

File: zetaclient/orchestrator/bootstrap.go
Comment (ws4charlie) [reply zetaclient/orchestrator/bootstrap.go:142]: It's just a renaming

File: zetaclient/config/types.go
Comment (gartnera) [reply zetaclient/config/types.go:136]: Ok yeah I think this logic is fine

File: changelog.md
Comment (lumtis) [inline changelog.md:54]: Why do we have some many new entries?

File: zetaclient/orchestrator/orchestrator_test.go
Comment (lumtis) [inline zetaclient/orchestrator/orchestrator_test.go:524]: Can we use switch case?

**Review by lumtis — COMMENTED** [review]
Let's wait for a review from @swift1337 for this one.

Not specifically for this PR but I think we should remove the concept of EVM and BTC configs in the future. Just keeping a list of config associated with a chain ID

File: changelog.md
Comment (ws4charlie) [reply changelog.md]: removed accidentally included change log entries during merge https://github.com/zeta-chain/node/pull/2870/commits/582d1de93c6f8531141eb15b0a3c952fc6eba8c5

File: zetaclient/orchestrator/orchestrator_test.go
Comment (ws4charlie) [reply zetaclient/orchestrator/orchestrator_test.go]: done in commit https://github.com/zeta-chain/node/pull/2870/commits/582d1de93c6f8531141eb15b0a3c952fc6eba8c5

**Comment by ws4charlie** [comment]
> Let's wait for a review from @swift1337 for this one.
> 
> Not specifically for this PR but I think we should remove the concept of EVM and BTC configs in the future. Just keeping a list of config associated with a chain ID

Totally agree. We'll need to a unified config struct (containing endpoint, user, password, etc.) for every external chain.

**Review by kingpinXD — APPROVED** [review]

File: cmd/zetaclientd/start_utils.go
Comment (swift1337) [inline cmd/zetaclientd/start_utils.go:59]: Can we use a package for this? https://github.com/showa-93/go-mask 

```go
type FooBar struct {
			ABC string   `mask:"filled"` // will be masked
}
```

File: zetaclient/config/types.go
Comment (swift1337) [inline zetaclient/config/types.go:96]: Can we make sure that `.BitcoinConfig` is not used anywhere except `GetBTCConfig(chainID)`?

File: cmd/zetaclientd/start_utils.go
Comment (ws4charlie) [reply cmd/zetaclientd/start_utils.go:59]: looks sweet. Let's use it.

File: zetaclient/config/types.go
Comment (ws4charlie) [reply zetaclient/config/types.go:96]: Yep. That's right.

File: cmd/zetaclientd/start_utils.go
Comment (ws4charlie) [reply cmd/zetaclientd/start_utils.go]: https://github.com/zeta-chain/node/pull/2870/commits/1b23e969da315ab2e8738d32b5fd0e2068f99358

**Review by gartnera — APPROVED** [review]

**Pull request merged**
